backend/de.metas.business/src/main/java/de/metas/shipping/PurchaseOrderToShipperTransportationService.java

```diff
@@ -252,4 +242,10 @@ public ReportResultData printSSCC18_Labels(@NonNull final OrderId orderId)
 		return result.getReportData();
 	}
 
+	public boolean deleteShippingPackagesForOrderIfPossible(@NonNull final OrderId orderId)
+	{
+		final Collection<I_M_ShipperTransportation> transportationOrders = shipperTransportationDAO.getTransportationOrdersAssignedToOrder(orderId);
+		return transportationOrders.stream().noneMatch(I_M_ShipperTransportation::isProcessed);
+		//TODO Adrian add delete logic
```

Comment:
IMHO you should
* get all M_ShippingPackage(s) by orderId(s)
* extract ShipperTransportationIds from them
* get all M_ShipperTransportation(s) by the ids extracted previously
* make sure they are not processed
   * in case some of them are processed why not throwning exception? what would be the reason to not do so?
* delete the M_ShippingPackage(s) retrieved